File: csharp/Handler.cs
using Amazon.DynamoDBv2;
using Amazon.Lambda.Core;
using Amazon.Lambda.APIGatewayEvents;
using System.Collections.Generic;
using System.Net;
using Amazon.DynamoDBv2.Model;
using System.Threading.Tasks;
using System.Linq;
using Newtonsoft.Json;

namespace AwsDotnetCsharp
{
    public class Handler
    {
        private const bool desc = true;

        [LambdaSerializer(typeof(Amazon.Lambda.Serialization.Json.JsonSerializer))]
        public async Task<APIGatewayProxyResponse> Hello(APIGatewayProxyRequest request, ILambdaContext context)
        {
            var clientConfig = new AmazonDynamoDBConfig()
            {
                RegionEndpoint = Amazon.RegionEndpoint.USWest2
            };

            var client = new AmazonDynamoDBClient(clientConfig);
            var query = new QueryRequest
            {
                TableName = System.Environment.GetEnvironmentVariable("DYNAMODB_TABLE"),
                IndexName = System.Environment.GetEnvironmentVariable("DYNAMODB_GSI_USERID_NOTEID"),
                KeyConditionExpression = "userId = :userId",
                ProjectionExpression="userId, noteId, notebook, #note_text",
                ExpressionAttributeNames = new Dictionary<string, string> {
                    {"#note_text", "text"}
                },
                ExpressionAttributeValues = new Dictionary<string, AttributeValue> {
                    {":userId", new AttributeValue { S =  "arbiter" }}
                },
                ScanIndexForward = desc
            };

            var cleaned = new Dictionary<string, AttributeValue>();

            QueryResponse results = null;
            results = await client.QueryAsync(query);
            foreach (var item in results.Items)
                item.ToList().ForEach(x => cleaned[x.Key] = x.Value);

            var headers = new Dictionary<string, string>();
            headers.Add("Content-Type", "application/json");

            return await Task.Run(() => new APIGatewayProxyResponse()
            {
                StatusCode = (int)HttpStatusCode.OK,
                Headers = headers,
                Body = JsonConvert.SerializeObject(new { results = cleaned })
            });
        }
    }
}

File: fsharp/Handler.fs
namespace AwsDotnetFsharp

open System.Collections.Generic
open System.Net
open Amazon.Lambda.Core
open Amazon.Lambda.APIGatewayEvents
open Amazon.DynamoDBv2
open Amazon.DynamoDBv2.Model
open Newtonsoft.Json

[<assembly:LambdaSerializer(typeof<Amazon.Lambda.Serialization.Json.JsonSerializer>)>]
do ()

module Handler =
    open System
    open System.IO
    open System.Text

    let hello(request : APIGatewayProxyRequest, context : ILambdaContext): APIGatewayProxyResponse =

        let clientConfig = new AmazonDynamoDBConfig()
        clientConfig.RegionEndpoint <- Amazon.RegionEndpoint.USWest2

        let client = new AmazonDynamoDBClient(clientConfig)
        let query = new QueryRequest()
        query.TableName <- System.Environment.GetEnvironmentVariable("DYNAMODB_TABLE")
        query.IndexName <- System.Environment.GetEnvironmentVariable("DYNAMODB_GSI_USERID_NOTEID")
        query.KeyConditionExpression <- "userId = :userId"
        query.ProjectionExpression <- "userId, noteId, notebook, #note_text"
        query.ScanIndexForward <- true

        let names = Dictionary<string, string>(dict [ ("#note_text", "text") ])
        query.ExpressionAttributeNames <- names

        let values = Dictionary<string, AttributeValue>(dict [ (":userId", new AttributeValue(S =  "arbiter")) ])
        query.ExpressionAttributeValues <- values

        let cleaned = new Dictionary<string, AttributeValue>()
        let mutable results = new QueryResponse()
        results <- client.QueryAsync(query)
        |> Async.AwaitTask
        |> Async.RunSynchronously

        for item in results.Items do
            item.Keys |> Seq.iter (fun k -> cleaned.[k] <- item.[k])

        let headers = new Dictionary<string, string>()
        headers.Add("Content-Type", "application/json")

        let response = new APIGatewayProxyResponse()
        response.StatusCode <- int HttpStatusCode.OK
        response.Headers <- headers

        let body = {| results=cleaned |}
        response.Body <- JsonConvert.SerializeObject(body)

        response
